### PR TITLE
Handle shopware version stability

### PR DESCRIPTION
**Pull request by nlubisch**
The shopware store API can't handle any suffixed version stability like "5.5.0-RC1".
> Ihre Shopware-Version wurde nicht richtig übermittelt oder ist in unserem System nicht bekannt.
> Error code: BinariesException-12

In order to make it possible to use the shopware store correctly with versions like that we need to cut the stability off.
I added the stability as `version_text`.

### CODE REVIEW

File: app/AppKernel.php [hunk 1]
```diff
@@ -91,23 +92,22 @@ private function loadReleaseFromComposer()
...
         try {
-            list($version, $sha) = explode('@', \PackageVersions\Versions::getVersion('shopware/shopware'));
+            list($version, $sha) = explode('@', Versions::getVersion('shopware/shopware'));
 
-            /*
-             * Trim leading v from versions like "v5.4.6"
-             */
-            $version = ltrim(strtolower($version), 'v');
+            // Split versions like "v5.5.0-RC1" into "v5.5.0" and "RC1"
+            list($strippedVersion, $stability) = explode('-', $version);
```
Comment (bcremer) [inline app/AppKernel.php]: So much string manipulation going on here.
Would't it be easier to use the regex`s match to get the already well defined version number.
No case and prefix manipulations necessary.

**Review by bcremer — CHANGES_REQUESTED** [review]

File: app/AppKernel.php (hunk 1 above)
Comment (nlubisch) [reply app/AppKernel.php]: I'll fix it :)

**Comment by nlubisch** [comment]
Updated and condensed version is now available :)

File: app/AppKernel.php (hunk 1 above)
Comment (hlohaus) [reply app/AppKernel.php]: try `#^v?(?P<version>\d+\.\d\.\d)(?:-(?P<version_text>[\d\w]+))$#`
i like regex :)

File: app/AppKernel.php [hunk 2]
```diff
@@ -91,23 +92,15 @@ private function loadReleaseFromComposer()
...
-                throw new OutOfBoundsException(sprintf('Version "%s" not in expected format', $version));
+            if (!preg_match('/^v?([\d]+\.[\d]+\.[\d]+)(\-([a-zA-Z\d]{0,4}))?$/', $version, $versionMatches)) {
+                throw new \OutOfBoundsException(sprintf('Version "%s" not in expected format', $version));
             }
 
-            $this->release['version'] = $version;
+            $this->release['version'] = $versionMatches[1];
             $this->release['revision'] = substr($sha, 0, 10);
-            $this->release['version_text'] = ''; // Feel free to change this
+            $this->release['version_text'] = $versionMatches[3] ?: '';
```
Comment (hlohaus) [inline app/AppKernel.php]: Notice: Undefined index :)

File: app/AppKernel.php [hunk 3]
```diff
@@ -91,23 +92,15 @@ private function loadReleaseFromComposer()
...
-             * Trim leading v from versions like "v5.4.6"
-             */
-            $version = ltrim(strtolower($version), 'v');
-
-            /*
-             * Make sure the version matches some expected patterns like "5.4.0" or "5.0.0-RC1"
-             */
-            if (!preg_match('/^([\d]+\.[\d]+\.[\d]+(\-[a-zA-Z\d]{0,4})?)$/', $version)) {
-                throw new OutOfBoundsException(sprintf('Version "%s" not in expected format', $version));
+            if (!preg_match('/^v?([\d]+\.[\d]+\.[\d]+)(\-([a-zA-Z\d]{0,4}))?$/', $version, $versionMatches)) {
```
Comment (hlohaus) [inline app/AppKernel.php]: `#^v?(?P<version>\d+\.\d\.\d)(?:-(?P<version_text>[\d\w]+))$#`
with group names :)

File: app/AppKernel.php (hunk 1 above)
Comment (oktupol) [reply app/AppKernel.php]:
```
#^v?(?P<version>\d+\.\d\.\d)(?:-(?P<version_text>[\d\w]+))$#
                              :-(
```
Why is your regex so sad? 🙁

File: app/AppKernel.php [hunk 4]
```diff
@@ -91,23 +92,15 @@ private function loadReleaseFromComposer()
         }
 
         try {
-            list($version, $sha) = explode('@', \PackageVersions\Versions::getVersion('shopware/shopware'));
+            list($version, $sha) = explode('@', Versions::getVersion('shopware/shopware'));
```
Comment (bcremer) [inline app/AppKernel.php]: That splitting could also be done inside the regex :man_shrugging: 
Also the regex should be case insensitive to be compatible with the code before.

Comment (nlubisch) [reply app/AppKernel.php]: Whoops, didn't saw that. I should buy me glasses :neutral_face:

**Comment by nlubisch** [comment]
Looks good to me on my first view. Will have a deeper look when i got some time :)

**Pull request closed without merging**